scipy/stats/tests/test_distributions.py

```diff
@@ -1496,6 +1511,13 @@ def test_fit_warnings(self):
         assert_raises(FitDataError, stats.pareto.fit, [5, 2, 3], floc=1,
                       fscale=3)
 
+    def test_negative_data(self):
+        # The purpose of this test is to make sure that no warnings are raised
+        # for all negative data.
```

Comment:
I don't remember - is it just if it's _all_ negative that we were worried?

Reply:
We discussed all negative data (since the original method of obtaining the upper bound was np.max(). It might not be needed but it is a bit of an extreme case that isn't tested elsewhere.